### PR TITLE
feat: metrics integration

### PR DESCRIPTION
**Pull request by danielhe4rt**
## Motivation

Since we're about to release the metrics base to our users at v1 implementation, it's good to have a base integration with tbp-consumer-api for now. 

This PR adds the request and population of this specific page inside the app.

vai tomar no cu react

Cheers for tw:gabrieldneto for helping me to implement it.

### CODE REVIEW

File: src/resources/components/stats/stats.tsx [hunk 1]
```diff
@@ -1,20 +1,55 @@
...
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+        ],
+        topCategories: [
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+        ],
+      };
+    }
```
Comment (moovimentos) [inline src/resources/components/stats/stats.tsx]: Would it be better to set times and messages to 0 and leave channels and categories empty?

File: src/resources/components/stats/stats.tsx [hunk 2]
```diff
@@ -1,20 +1,55 @@
...
-    ],
-  };
+  const [authorization] = useStorage<AccessTokenResponse>("accessToken");
+
+  const [stats, setStats] = useState<MetricsResponse>();
+
+  useEffect(() => {
+    getMetrics(authorization?.access_token)
+      .then((res) => {
+        console.log(res);
```
Comment (moovimentos) [inline src/resources/components/stats/stats.tsx]: Remove the debug

File: src/scripting/index.ts [hunk 3]
```diff
@@ -97,6 +101,7 @@ export default class Kernel {
       }
       // @ts-ignore
       interval = setInterval(() => {
+        console.log(currentCategory);
```
Comment (moovimentos) [inline src/scripting/index.ts]: Remove the debug

**Review by moovimentos — CHANGES_REQUESTED** [review]

File: src/resources/components/stats/stats.tsx [hunk 4]
```diff
@@ -1,20 +1,55 @@
...
+      topChannels: stats.user_metrics_by_channel.slice(0, 4).map((channel) => {
+        return `https://twitch-cdn.danielheart.dev/u/${channel.channel_id}.png`;
+      }),
+      topCategories: stats.user_metrics_by_category
+        .slice(0, 4)
+        .map((channel) => {
+          return `https://twitch-cdn.danielheart.dev/c/${channel.category_id}.png`;
+        }),
+    };
+  }, [stats]);
```
Comment (moovimentos) [inline src/resources/components/stats/stats.tsx]: I got this when I tried to open the stats
image

I printed `console.log(statsData.topChannels)`
image

**Review by moovimentos — CHANGES_REQUESTED** [review]

File: src/resources/components/stats/stats.tsx (hunk 4 above)
Comment (moovimentos) [reply src/resources/components/stats/stats.tsx]: return of stats
image

File: src/resources/components/stats/stats.tsx [hunk 5]
```diff
@@ -1,20 +1,55 @@
...
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+        ],
+      };
+    }
+    return {
+      hoursWatched: stats.main_metrics.minutes_watched
+        ? stats.main_metrics.minutes_watched * 60
```
Comment (moovimentos) [inline src/resources/components/stats/stats.tsx]: It should be divided (/) instead of multiplied (*)?

**Review by moovimentos — CHANGES_REQUESTED** [review]

File: src/resources/components/stats/stats.tsx (hunk 4 above)
Comment (danielhe4rt) [reply src/resources/components/stats/stats.tsx]: This is related to the throttle API which were fixing here: https://github.com/basementdevs/tbp-consumer-api/pull/4

Soon I'll clear the rankings and you will be able to test it again.

File: src/resources/components/stats/stats.tsx (hunk 5 above)
Comment (danielhe4rt) [reply src/resources/components/stats/stats.tsx]: My name is potato and I skipped math classes.

File: src/resources/components/stats/stats.tsx (hunk 1 above)
Comment (danielhe4rt) [reply src/resources/components/stats/stats.tsx]: Done.

**Review by moovimentos — APPROVED** [review]
If his nose is that big, imagine this man's mind!
LGTM!

**Pull request merged**